### PR TITLE
Docs: Minor edits to the README and several md files

### PR DESCRIPTION
**Pull request by bharjala**
Minor edits:
- Update README: Capitalized the G and S in "Getting Started," and moved "guide" to match the section title in the docs.
- Fixed sentence structure: Changed "download" to "grafana.com/get" and changed "get" to "download."
- Docs: Replace "datasources" with "data sources."

### CODE REVIEW

File: docs/sources/permissions/overview.md [hunk 1]
```diff
@@ -34,9 +34,9 @@ in that organization. Learn more about [Organization Roles]({{< relref "permissi
...
 
-## Datasource Permissions
+## Data Source Permissions
 
-Per default, a datasource in an organization can be queried by any user in that organization. For example a user with `Viewer` role can still
+Per default, a data source in an organization can be queried by any user in that organization. For example a user with `Viewer` role can still
 issue any possible query to a data source, not just those queries that exist on dashboards he/she has access to.
 
-Datasource permissions allows you to change the default permissions for datasources and restrict query permissions to specific **Users** and **Teams**. Read more about [Datasource Permissions]({{< relref "permissions/datasource_permissions.md" >}}).
+Data source permissions allows you to change the default permissions for data sources and restrict query permissions to specific **Users** and **Teams**. Read more about [Data Source Permissions]({{< relref "permissions/datasource_permissions.md" >}}).
```
Comment (marcusolsson) [inline docs/sources/permissions/overview.md]:
```
Read more about [data source permissions](...
```

File: docs/sources/plugins/developing/auth-for-datasources.md [hunk 2]
```diff
@@ -7,25 +7,25 @@ parent = "developing"
 weight = 3
 +++
 
-# Authentication for Datasource Plugins
+# Authentication for Data Source Plugins
```
Comment (marcusolsson) [inline docs/sources/plugins/developing/auth-for-datasources.md]: Sentence case

File: docs/sources/plugins/developing/backend-plugins-guide.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ The backend plugin feature is implemented with the [HashiCorp plugin system](htt
 - Plugins are easy to develop: just write a Go application and `go build` (or use any other language which supports gRPC).
 - Plugins can be relatively secure: The plugin only has access to the interfaces and args given to it, not to the entire memory space of the process.
 
-## Datasource Plugin Interface
+## Data Source Plugin Interface
```
Comment (marcusolsson) [inline docs/sources/plugins/developing/backend-plugins-guide.md]: Sentence case

File: docs/sources/plugins/developing/backend-plugins-guide.md [hunk 4]
```diff
@@ -111,7 +111,7 @@ type JsonDatasource struct {
 }
 ```
 
-The only requirement for the plugin type is that it should extend `plugin.NetRPCUnsupportedPlugin`. You can include more fields into your struct if you want to add some datasource-specific features, like logging, cache etc:
+The only requirement for the plugin type is that it should extend `plugin.NetRPCUnsupportedPlugin`. You can include more fields into your struct if you want to add some data-source-specific features, like logging, cache etc:
```
Comment (marcusolsson) [inline docs/sources/plugins/developing/backend-plugins-guide.md]: I'm not sure what the correct practice for this case. Would it be more correct to write `data source-specific` here?

File: docs/sources/plugins/developing/datasources.md [hunk 5]
```diff
@@ -8,17 +8,17 @@ parent = "developing"
 weight = 5
 +++
 
-# Datasources
+# Data Sources
```
Comment (marcusolsson) [inline docs/sources/plugins/developing/datasources.md]: Sentence case

File: docs/sources/plugins/developing/datasources.md [hunk 6]
```diff
@@ -38,15 +38,15 @@ There are two datasource specific settings for the plugin.json
 
 These settings indicates what kind of data the plugin can deliver. At least one of them have to be true
 
-## Datasource
+## Data Source
```
Comment (marcusolsson) [inline docs/sources/plugins/developing/datasources.md]: Sentence case

File: docs/sources/plugins/developing/plugin-review-guidelines.md [hunk 7]
```diff
@@ -155,7 +155,7 @@ A basic guide for data sources can be found [here](http://docs.grafana.org/plugi
 
 If possible, any passwords or secrets should be be saved in the `secureJsonData` blob. To encrypt sensitive data, the Grafana server's proxy feature must be used. The Grafana server has support for token authentication (OAuth) and HTTP Header authentication. If the calls have to be sent directly from the browser to a third-party API then this will not be possible and sensitive data will not be encrypted.
 
-Read more here about how [Authentication for Datasources]({{< relref "auth-for-datasources.md" >}}) works.
+Read more here about how [Authentication for Data Sources]({{< relref "auth-for-datasources.md" >}}) works.
```
Comment (marcusolsson) [inline docs/sources/plugins/developing/plugin-review-guidelines.md]: This should probably be:

```
about how [authentication for data sources](...
```
```

File: docs/sources/reference/templating.md [hunk 8]
```diff
@@ -147,7 +147,7 @@ Type | Description
 ------- | --------
 *Query* | This variable type allows you to write a data source query that usually returns a list of metric names, tag values or keys. For example, a query that returns a list of server names, sensor ids or data centers.
 *Interval* | This variable can represent time spans. Instead of hard-coding a group by time or date histogram interval, use a variable of this type.
-*Datasource* | This type allows you to quickly change the data source for an entire Dashboard. Useful if you have multiple instances of a data source in for example different environments.
+*Data Source* | This type allows you to quickly change the data source for an entire Dashboard. Useful if you have multiple instances of a data source in for example different environments.
```
Comment (marcusolsson) [inline docs/sources/reference/templating.md]: Lowercase S, I think

File: public/app/plugins/datasource/graphite/README.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
-# Graphite Datasource -  Native Plugin
+# Graphite Data Source -  Native Plugin
```
Comment (marcusolsson) [inline public/app/plugins/datasource/graphite/README.md]: For the READMEs, let's keep Title Case for now. This is fine.

**Review by marcusolsson — CHANGES_REQUESTED** [review]

**Comment by marcusolsson** [comment]
Thanks for this! 👍 

Actually, don't bother with the casing for now. We can do that in subsequent PRs. Let's leave this one as a search-and-replace. 

Only real issue is the YAML example should be corrected.

File: docs/sources/features/datasources/cloudwatch.md [hunk 10]
```diff
@@ -251,17 +251,17 @@ it costs $0.01 per 1,000 GetMetricStatistics or ListMetrics requests. For each q
...
 
-Here are some provisioning examples for this datasource.
+Here are some provisioning examples for this data source.
 
 Using a credentials file
 ```yaml
 apiVersion: 1
 
-datasources:
+data sources:
```
[the remark this answers is not in the document]
Comment (bharjala) [reply docs/sources/features/datasources/cloudwatch.md]: Got it! Will not touch yaml examples.

File: docs/sources/plugins/developing/plugin-review-guidelines.md (hunk 7 above)
Comment (bharjala) [reply docs/sources/plugins/developing/plugin-review-guidelines.md]: Makes sense!

File: public/app/plugins/datasource/graphite/README.md (hunk 9 above)
Comment (bharjala) [reply public/app/plugins/datasource/graphite/README.md]: Thanks, noted.

File: docs/sources/features/datasources/azuremonitor.md [hunk 11]
```diff
@@ -31,7 +31,7 @@ The data source can access metrics from four different services. You can configu
 - [Guide to setting up an Azure Active Directory Application for Azure Log Analytics.](https://dev.loganalytics.io/documentation/Authorization/AAD-Setup)
 - [Quickstart Guide for Application Insights.](https://dev.applicationinsights.io/quickstart/)
 
-1. Accessed from the Grafana main menu, newly installed data sources can be added immediately within the Data Sources section. Next, click the "Add data source" button in the upper right. The Azure Monitor data source will be available for selection in the Cloud section in the list of data sources.
+1. Accessed from the Grafana main menu, newly installed data sources can be added immediately within the data sources section. Next, click the "Add data source" button in the upper right. The Azure Monitor data source will be available for selection in the Cloud section in the list of data sources.
```
Comment (marcusolsson) [inline docs/sources/features/datasources/azuremonitor.md]: This refers to the section in the main menu, where it says "Data Sources", so we should leave this one as is.

File: docs/sources/permissions/datasource_permissions.md [hunk 12]
```diff
@@ -10,11 +10,11 @@ parent = "permissions"
 weight = 4
 +++
 
-# Data Source Permissions
+# Data Source Permissions API
```
Comment (marcusolsson) [inline docs/sources/permissions/datasource_permissions.md]: This page describes concept of data source permissions, and not the API (as opposed to docs/sources/http_api/datasource_permissions.md).

File: docs/sources/plugins/developing/datasources.md [hunk 13]
```diff
@@ -72,10 +72,10 @@ Request object passed to datasource.query function:
 }
 ```
 
-There are two different kinds of results for datasources;
-time series and table. Time series is the most common format and is supported by all datasources and panels. Table format is only supported by the InfluxDB datasource and table panel. But we might see more of this in the future.
+There are two different kinds of results for data sources;
+time series and table. Time series is the most common format and is supported by all data sources and panels. Table format is only supported by the InfluxDB data source and table panel. But we might see more of this in the future.
 
-Time series response from datasource.query.
+Time series response from data source.query.
```
Comment (marcusolsson) [inline docs/sources/plugins/developing/datasources.md]: This is referring to a code function, and should be left as is.

File: docs/sources/permissions/datasource_permissions.md [hunk 14]
```diff
@@ -10,44 +10,44 @@ parent = "permissions"
 weight = 4
 +++
 
-# Datasource Permissions
+# Data Source Permissions API
 
-> Datasource Permissions is only available in Grafana Enterprise. Read more about [Grafana Enterprise]({{< relref "enterprise" >}}).
+> The Data Source Permissions API is only available in Grafana Enterprise. Read more about [Grafana Enterprise]({{< relref "enterprise" >}}).
```
Comment (marcusolsson) [inline docs/sources/permissions/datasource_permissions.md]: Same thing here.

File: docs/sources/permissions/datasource_permissions.md [hunk 15]
```diff
@@ -10,44 +10,44 @@ parent = "permissions"
...
 
-Datasource permissions allows you to restrict access for users to query a datasource. For each datasource there is
+The Data Source Permissions API allows you to restrict access for users to query a data source. For each data source there is
```
Comment (marcusolsson) [inline docs/sources/permissions/datasource_permissions.md]: ... and here

**Review by marcusolsson — APPROVED** [review]

**Comment by marcusolsson** [comment]
Thanks a lot! This is a major improvement! 👍

**Pull request merged**